### PR TITLE
[SPARK-40949][CONNECT][PYTHON] Implement `DataFrame.sortWithinPartitions`

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?
Implement `DataFrame.sortWithinPartitions`

### Why are the changes needed?
for api coverage

### Does this PR introduce _any_ user-facing change?
yes, new method

### How was this patch tested?
added UT

### CODE REVIEW

File: connector/connect/src/main/protobuf/spark/connect/relations.proto [hunk 1]
```diff
@@ -168,6 +168,7 @@ message Aggregate {
 message Sort {
   Relation input = 1;
   repeated SortField sort_fields = 2;
+  bool is_global = 3;
```
Comment (zhengruifeng) [inline connector/connect/src/main/protobuf/spark/connect/relations.proto:171]: unfortunately, `global` is a Python keyword, so name it as `is_global`

**Review by HyukjinKwon — APPROVED** [review]

**Comment by zhengruifeng** [comment]
the 3 force pushes only resolved conflicts in generated files, let me merge it now

**Comment by zhengruifeng** [comment]
merged into master, thanks @HyukjinKwon

**Comment by amaliujia** [comment]
Late LGTM thanks

**Pull request closed without merging**